studio/components/interfaces/App/CommandMenuWrapper.tsx

```diff
@@ -31,6 +36,18 @@ const CommandMenuWrapper = observer(({ children }: PropsWithChildren<{}>) => {
     service: settings?.autoApiService?.serviceApiKey ?? undefined,
   }
 
+  const { data } = useEntityDefinitionsQuery(
+    {
+      projectRef: ui.selectedProject?.ref,
+      connectionString: ui.selectedProject?.connectionString,
+    },
+    { enabled: isOptedInToAI }
+  )
+  const cmdkMetadata = {
+    definitions: (data ?? []).map((def) => def.sql.trim()).join('\n\n'),
+    flags: { allowCMDKDataOptIn },
+  }
```

Comment:
can this be wrapped with a `useMemo()` because there's a `.map()` call. It might get really slow if the user has a lot of tables.